internal/services/containers/container_registry_cache_rule_resource.go

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package containers
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2023-07-01/cacherules"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/validate"
+	containerValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+)
+
+var _ sdk.Resource = ContainerRegistryCacheRule{}
+
+type ContainerRegistryCacheRule struct{}
+
+func (ContainerRegistryCacheRule) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:        pluginsdk.TypeString,
+			Required:    true,
+			Description: "The name of the cache rule.",
+		},
+
+		"registry": {
```

Comment:
I would opt for `container_registry_id` here, which contains both the resource group name and registry name, as input.

How to work with that ID is shown here: https://github.com/hashicorp/terraform-provider-azurerm/blob/46a95c1b799aca661d980992e65779e8576178e7/internal/services/containers/container_registry_task_resource.go#L126